src/game/shared/swarm/asw_gamerules.cpp

```diff
@@ -9414,6 +9442,12 @@ void CAlienSwarm::LevelInitPostEntity()
 	}
 	// todo: if we fail to load the campaign save file above, then gracefully fall into single mission mode?
 
+	// restarting on the first mission of the campaign will not fail outstanding execution
+	if ( GetCampaignSave() && GetCampaignInfo() && GetCampaignInfo()->Missions.Count() > 1 && !V_strcmp( GetCampaignInfo()->Missions[1].MapName, STRING( gpGlobals->mapname ) ) )
```

Comment:
`GetCampaignSave()` and `GetCampaignInfo()` can both be cached in local variables